Host dashboard: Add "Record as paid" button for PayPal

flickz opened the pull request:
This PR follows up https://github.com/opencollective/opencollective/issues/1976

The goal is to add "Record as paid" button for expense with paypal payment method.

<img width="581" alt="Screenshot 2019-10-07 at 4 05 20 PM" src="https://user-images.githubusercontent.com/15707013/66329667-cbe01480-e926-11e9-8fbc-d3c65875b693.png">

#### TODO

- [ ] Modify [`payExpense`](https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/mutations/expenses.js#L340) mutation to allow recording of expense with paypal method

znarf commented:
@flickz what's the update on that? why is it WIP, what else needs to be done?

znarf commented:
@flickz https://github.com/opencollective/opencollective-frontend/pull/2800 is merged now

znarf commented:
@flickz this PR is stale, it will be close in 7 days if no activity

flickz commented:
The current failing test is a tricky one, it is not related to this PR but as a result of dependency update from https://github.com/opencollective/opencollective-frontend/commit/d866a8eed0fbeff793a0df0a345ea008c1644445 

I fixed it locally by updating my dependencies but I'm not sure how to achieve that in circle ci.

znarf commented:
@flickz I don't get why the title-case update would impact this PR only and not elsewhere.

Betree commented on components/expenses/Expense.js:
```diff
@@ -438,15 +445,16 @@ class Expense extends React.Component {
                     />
                   )}
-                  <div className="expenseActions" data-cy="expense-actions">
+                  <ExpenseActionsWrapper className="expenseActions" data-cy="expense-actions">
```
Not a requested change: `ExpenseActionsWrapper` doens't seems necessary, you can achieve the same effect by using `Flex`. Also isn't that an opportunity to get rid of the `expenseActions` class?

Betree commented on package-lock.json:
```diff
@@ -7308,7 +7308,7 @@
         },
```
Let's not commit package-lock

flickz commented:
> I don't get why the title-case update would impact this PR only and not elsewhere.

@znarf I'm not sure why to, but it started happening after I rebased, then I fixed it locally by updating my dependencies but this is the error below.


<img width="1056" alt="Screenshot 2019-12-04 at 11 00 34 AM" src="https://user-images.githubusercontent.com/15707013/70132991-91c09400-1685-11ea-969f-e867a23fe97d.png">

znarf commented:
@flickz this is not the error I see in the E2E tests, am I missing the link?

znarf commented:
And even in development, `titleCase` should not be a problem, make sure to rebase, `npm install` and delete your `node_modules` if necessary.

flickz commented:
I think the issue I'm having with this, the particular failing test is passing locally everytime I run it. And the changes in this PR are not linked to the failing test.

znarf commented:
It's fetching the API from the `enhancement/expense-paypal` branch, is this what you want? https://circleci.com/gh/opencollective/opencollective-frontend/61561

flickz commented:
We can change that to master now because we've merged the API and I'm locally testing with API master branch.
